server-spi/src/main/java/org/keycloak/storage/user/UserQueryProvider.java

```diff
@@ -323,14 +361,15 @@ default int getUsersCount(RealmModel realm, boolean includeServiceAccount) {
     }
 
     /**
-     * Get users that belong to a specific group. Implementations do not have to search in UserFederatedStorageProvider
+     * Get users that belong to a specific group. Implementations do not have to search in {@code UserFederatedStorageProvider}
```

Comment:
`Implementations do not have to search in {@code UserFederatedStorageProvider}` part should clarified / reworded

Reply:
To be honest, I would probably completely remove this part, as it basically holds for all methods that they should not do anything with federated storage. Maybe it should be in the documentation, that when you are implementing UserProvider for your storage you should return only users, that are part of your storage.

Reply:
I agree, class javadoc would be best place to put this information